### ISSUE TITLE
In Excel Client windows, error Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getFields'). This just started happening today.

### ISSUE THREAD

**Issue by jimbarrett33** [OfficeDev/office-js#4180]
Starting today, an error occurs when office.js loads in Excel client when the add-in loads in the task pane.

![image](https://github.com/OfficeDev/office-js/assets/8396655/5df0d746-69d4-449d-beb9-691549ea86b4)

This is the line of code:
` , k = oteljs.Contracts.Office.System.SDX.getFields("SDX", i)`

I don't think it occurs on the web. I noticed it started happening today with my add-in but I created a new simple add-in using Visual Studio Create Project template and it occurs there too.

It doesn't appear to impact the add-in successfully continuing to run/function but I haven't done a lot of testing around this.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: Microsoft® Excel® for Microsoft 365 MSO (Version 2401 Build 16.0.17231.20236) 64-bit 
* Operating System: Windows 10
* Browser (if using Office on the web): ______
 
## Expected behavior
There should not be an error in the console on startup. It seems as though, office.js is calling something on load that isn't created yet.

## Current behavior
As detailed above.

## Steps to reproduce
1. In Visual Studio 2022, create a new Excel Add-in using the "create project" template
2. Run the project
3. When the add-in loads in the Excel client, right-click and "Inspect" with the debug tool
4. Notice the error in the console window.

## Context
I have a subscription-based add-in in AppSource and I want to make sure it continues to function correctly for my customers.

**Comment by junmt** [comment]
When starting the add-in in Excel 2019, other scripts stop working due to this error and the add-in cannot be used.

**Comment by xai-msft** [comment]
Hi @jimbarrett33 and @junmt ,

Thank you for reporting the issue. We are currently investigating it.

**Comment by xai-msft** [comment]
@junmt Could you please let us know which add-in you are using? Have you been using them normally on Excel 2019 in the past?

**Comment by junmt** [comment]
@xai-msft 
Thank you for checking. The add-in I'm using is "F-RevoCRM Data Connector".
In Excel 2019, the '始める' button cannot be clicked.

**Comment by manaunl** [comment]
For Outlook add-in we experience the same issue

**Comment by zhenhuangMSFT** [comment]
@exextoc please note this issue

**Comment by bnayak-conga** [comment]
For word add-in also we are facing same issue

**Comment by xai-msft** [comment]
For the 'getField' issue, we are working on a fix. Internal tracking number: 8784639

**Comment by xai-msft** [comment]
@junmt For the "F-RevoCRM Data Connector", since the taskpane is normally loaded, and just that the button '始める' is unclickable, it might be an issue with this specific add-in. Please consider reach out to add-in's developer/technical support to report the issue.

**Comment by bnayak-conga** [comment]
Any ETA for this?

Thanks,
Bhagya Nayak


Increase revenue lifecycle certainty
Leave a review
|Support|Twitter|
LinkedIn

**Comment by deepak-spotdraft** [comment]
Is there a way to circumvent this issue? We are facing a production outage because of this

**Comment by Jayzh7** [comment]
Hi @deepak-spotdraft, may I ask what issues exactly are you having that caused the production outage?

**Comment by deepak-spotdraft** [comment]
Just this error, which has started coming up.. The plugin just won't load

**Comment by zhenhuangMSFT** [comment]
Hi @deepak-spotdraft, thanks for providing extra details. Does it happen only on Excel 2019 or also on Office of new versions?

**Comment by deepak-spotdraft** [comment]
Hi @zhenhuangMSFT , I should have clarified, ours is a Word Addin, using Office js,

```
main.3527fa813b6722f8.js:1 Unhandled Promise rejection: Cannot read properties of undefined (reading 'getFields') ; Zone: <root> ; Task: Promise.then ; Value: TypeError: Cannot read properties of undefined (reading 'getFields')
    at b.create (office.js:76:16542)
    at a (office.js:76:17631)
    at office.js:76:17762
    at v.invoke (polyfills.6724dd477cea35b3.js:1:6563)
    at M.run (polyfills.6724dd477cea35b3.js:1:1959)
    at polyfills.6724dd477cea35b3.js:1:16760
    at v.invokeTask (polyfills.6724dd477cea35b3.js:1:7181)
    at M.runTask (polyfills.6724dd477cea35b3.js:1:2576)
    at _ (polyfills.6724dd477cea35b3.js:1:9198) TypeError: Cannot read properties of undefined (reading 'getFields')

```

**Comment by willehpeh** [comment]
I suspect I have a similar issue, developing an Excel addin that has had no issues previously:

TypeError: undefined is not an object (evaluating 'oteljs.Contracts.Office.System.SDX.getFields')

**Comment by bchenhs** [comment]
Hi @exextoc  @zhenhuangMSFT We are seeing the exact same error `undefined is not an object (evaluating 'oteljs.Contracts.Office.System.SDX.getFields')` on our Outlook add-in. This appears to affect our users on Outlook for Mac. It started at around Feb 26 11 PM UTC.

**Comment by zhenhuangMSFT** [comment]
Hi @deepak-spotdraft, could you share the repro steps? Which version of Office are you on? what add-in are you using? That could help us better investigate. Thanks!

**Comment by willehpeh** [comment]
@zhenhuangMSFT has this JS file been updated in the past 24-48 hours?

https://appsforoffice.microsoft.com/lib/1/hosted/office.js

**Comment by janbensch** [comment]
Same issue is now appearing in our Outlook-Add-In project, which is still in development. Cleared browser and Outlook Cache about 10 Minutes ago. Before then, everything worked fine.

Environment:
- Outlook 2021 OnPremise
- Exchange 2019 OnPremise
- office.js imported via CDN

**Comment by milannakum** [comment]
The error is also there for Office 2019 Powerpoint.

**Comment by bebe0806** [comment]
Confirmed Outlook on WEB:

office.js:76  Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getFields')
    at b.create (office.js:76:16542)
    at a (office.js:76:17631)
    at office.js:76:17762

**Comment by zhenhuangMSFT** [comment]
Thank all for reporting the issue! This is a potential regression with the latest released office.js. We are working on the fix. Will update here if any news. Thanks again!

**Comment by aplahore4** [comment]
![issue](https://github.com/OfficeDev/office-js/assets/81356018/ce34e790-d488-420e-8892-84ccb8582953)
Hi @exextoc @zhenhuangMSFT 
I am facing following two issues along with the stated by every one

1. `Access to fetch at 'https://excel-telemetry.officeapps.live.com/xt/RemoteUls.ashx?waccluster=PSG3&officeserverversion=16.0.17424.42305&usid=1ea3ae8d-2cc3-47b3-8d52-396251da0320' from origin 'https://excel.officeapps.live.com' has been blocked by CORS policy: Request header field x-nouls is not allowed by Access-Control-Allow-Headers in preflight response.`

2. `EwaTS.js:3441 POST https://excel-telemetry.officeapps.live.com/xt/RemoteUls.ashx?waccluster=PSG3&officeserverversion=16.0.17424.42305&usid=1ea3ae8d-2cc3-47b3-8d52-396251da0320 net::ERR_FAILED`

**Comment by wtravO** [comment]
> Is there a way to circumvent this issue? We are facing a production outage because of this

Also facing a production outage because of this.

**Comment by willehpeh** [comment]
Can you roll back to the previous version of office.js in the meantime?

**Comment by ADearingIntlFarming** [comment]
> > Is there a way to circumvent this issue? We are facing a production outage because of this
> 
> Also facing a production outage because of this.

Same, tagging for updates

**Comment by billzeman** [comment]
We see this also. possibly related to the changes that cause this IE11 issue:
https://github.com/OfficeDev/office-js/issues/4181
is there an archive anywhere of the previous version so we can get our customers working again?

**Comment by daviddykstrajambo** [comment]
We are also having this issue with our Outlook add-in. However we are only having the problem in the New Outlook on Windows, and on the web. Also on Mac.

**Comment by zacfarnsworth** [comment]
Hi everyone, so sorry about the issues you are facing! We have initiated a rollback of this update to office-js. It is in progress and I will update this page when it has completely rolled back.

**Comment by zacfarnsworth** [comment]
We completed the rollback to a stable version late last night. Please let me know if issues persist.

**Comment by baymatstech** [comment]
Thank you @zacfarnsworth - Working here again

**Comment by aplahore4** [comment]
@zacfarnsworth Mine started working thanks!

**Comment by jimbarrett33** [comment]
@zacfarnsworth all good here too. Thank you!

Could you give us a heads up on the next attempt to deploy that code so we can watch out?

Or is this a place/blog we can monitor with updates to office-js?

**Comment by bebe0806** [comment]
Would be nice if you could update also the npm package.

**Comment by chandpravandan** [comment]
closing this since the mentioned issue was fixed

please feel free to create a new issue if you experience this scenario
